src/lib/ConfirmationController.js

```diff
@@ -1,146 +1,70 @@
 export default class ConfirmationController {
-  #originalSubmitter
+  #resolve
+  #dialogSelector
   #initialContent
-  #config = {
-    dialogSelector: '#confirm',
-    activeClass: 'modal--active',
-    acceptSelector: '#confirm-accept',
-    denySelector: '.confirm-cancel',
-    animationDuration: 300,
-    showConfirmCallback: null,
-    contentSlots: {
-      title: {
-        contentAttribute: 'turbo-confirm',
-        slotSelector: '#confirm-title'
-      },
-      body: {
-        contentAttribute: 'confirm-details',
-        slotSelector: '#confirm-body'
-      },
-      acceptText: {
-        contentAttribute: 'confirm-button',
-        slotSelector: '#confirm-accept'
-      }
-    }
-  }
 
-  constructor(options = {}) {
-    // override default config with any options passed in
-    for(const [key, value] of Object.entries(options)) {
-      this.#config[key] = value
-    }
-    // bind methods for use as event listeners
+  constructor(selector, delegate) {
+    this.#dialogSelector = selector
+    this.delegate = delegate
+
     this.accept = this.accept.bind(this)
     this.deny = this.deny.bind(this)
   }
 
-  perform(event) {
-    // If present, the confirmation dialog was already accepted.
-    // We just need to clean up and the original action will proceed as normal.
-    if (this.#originalSubmitter) return this.#teardown()
-    // Otherwise, we need to prevent the default action
-    event.preventDefault()
-    // and store a reference to the clicked element
-    this.#originalSubmitter = this.#clickTarget(event)
-    // and show the confirmation dialog
-    this.#showConfirm()
+  showConfirm(contentMap, resolver) {
+    this.#resolve = resolver.bind(this)
+    if (!this.#initialContent) this.#initialContent = this.element.innerHTML
+
+    for(const [selector, content] of Object.entries(contentMap)) {
+      const target = this.element.querySelector(selector)
+      if (target && content) {
+        target.innerHTML = content
+      }
+    }
+
+    this.#setupListeners()
+    this.delegate.showConfirm(this.element)
   }
 
   accept() {
-    // re-trigger the original action
-    this.#originalSubmitter.click()
+    this.#resolve(true)
+    this.#teardown()
   }
 
   deny() {
+    this.#resolve(false)
```

Comment:
Thoughts on rejecting the promise on a cancel?
Benefit would be in a traditional Promise format it would look a lot nicer since I assume most cases don't need to handle a rejected case:
```js
tc.confirm('Are you sure?').then(()=> { /* accepted */ })
```

A downside is you've got to wrap it in a `try` in an `await` scenario
```js
try {
  await tc.confirm('Are you sure?')
  /* Accepted */
}
```

Reply:
@justwiebe I considered this also.  My instinct is not to use exceptions for control flow.  The Promise API seems to suggest doing so though..